### DISCUSSION TITLE
No genetics output despite calling OutIntGenetic > 0

### DISCUSSION THREAD

**Discussion by celinetree** [RangeShifter/RangeshiftR-tutorials#25] [Q&A] [answered]
Hello,

I am trying to run a simulation that will output genetic data, but for some reason RangeShiftR only generates the output file with column names with no data:
[Batch1_Sim1_Land1_Rep1_Genetics.txt](https://github.com/RangeShifter/RangeshiftR-tutorials/files/8287687/Batch1_Sim1_Land1_Rep1_Genetics.txt)

I set OutIntGenetic to both 1 and 5 with no luck. I also tried setting OutGenType equal to 1. The settings for my genetics module can be found in the sp2.txt script.

Individual, population, and the SMS heat map data are populated unlike the genetics output. I was wondering if perhaps someone could help me look over my script and parameter file to see if we can catch anything that may be stopping genetic data to be generated.
[Batch1_Sim1_Land1_Parameters.txt](https://github.com/RangeShifter/RangeshiftR-tutorials/files/8287732/Batch1_Sim1_Land1_Parameters.txt)
[sp2.txt](https://github.com/RangeShifter/RangeshiftR-tutorials/files/8287750/sp2.txt)

Please let me know if I can provide any more info.

- Celine

**Comment by annekathleenmalchow** [comment]
Hi Celine,

thanks for providing the output files with your question. I can see from the Parameters file, line 106, that there are no variable traits set. When you add the Genetics module to your RangeShiftR simulation, this only means that the individuals will be generated with a genome. But in order to link the genotype to a phenotype, i.e. expressing certain trait values, you have to set the chosen trait to be varying among individuals. You can do this by setting IndVar = TRUE for one of the dispersal traits, e.g. for emigration probability. For each variable trait, you need to provide the mean and standard deviation of its initial distribution (instead of only the constant value in case that IndVar = FALSE). Please see the help page of the respective dispersal sub-module for instructions how to do that. You can find an example in our [Tutorial 4](https://rangeshifter.github.io/RangeshiftR-tutorials/tutorial_4.html), section 2.3.

I hope this helps. Please get back to us here if you have further questions.
Anne

**Reply by aurcoul** [reply]
Thanks Anne for your answer. I am working with Céline on this project. One important piece of information she forgot to mention is that we want to simulate neutral loci (neutral landscape genetics study). We thought, in this case, IndVar must be set to FALSE (the loci are not related to dispersal traits). But obviously there is something we missed...
Aurélie.

**Comment by annekathleenmalchow** [comment]
Hi Aurélie,
For neutral landscape dynamics, you need to provide more detailed information about your genetic architecture via a special file. To do this, use the option `Architecture = 1` in the genetics module and provide the filename in `ArchFile`. (The simpler, pre-defined option `Architecture = 0` creates an architecture where there is one chromosome per trait, so that the genome is empty if all `IndVar=FALSE`.) Please consult our [User manual](https://github.com/RangeShifter/RangeShifter-software-and-documentation/blob/master/RangeShifter_v2.0_UserManual.pdf) for more information on the flexible genetic architecture, specifically sections 2.6.1, 2.6.3, and 3.2.8 with Fig. 3.23. If you then still have questions, feel free to post them here.
Cheers,
Anne

**Reply by aurcoul** [reply]
Ok, many thanks, we are investigating those options.

**Comment by celinetree** [comment]
Hello Anne,

My group and I have finalized our code using your suggestions. However, we get an error associated with the ArchFile when we run the simulation. Here is the output from running the RunRS() function:

>Checking Control parameters 

>Control Parameters checked

>Run Simulation(s) with random seed ...
>LandscapeFile headers OK: >/Users/celinecarneiro/Documents/Tempoland/RangeShiftR_out/Inputs/Maps_Control.txt
>PatchFile headers OK: /Users/celinecarneiro/Documents/Tempoland/RangeShiftR_out/Inputs/Patches.txt
>Checking Archiecture file frogarch.txt
>Error re-opening Architecture file with state 4

>Error in reading parameter file(s)... see RS log.

>***** Elapsed time: 1 seconds

>*****
>***** Simulation completed 
>***** Outputs folder: /Users/celinecarneiro/Documents/Tempoland/RangeShiftR_out/Outputs/
>*****
>$Errors
>[1] 0



I'm not sure what it doesn't like about the ArchFile we have. I've attached our ArchFile as well as the updated script and RS log. Any ideas? We are simulating neutral genetics so I assumed that we did not need a 'Trait' line.

[frogarch.txt](https://github.com/RangeShifter/RangeshiftR-tutorials/files/8312535/frogarch.txt)
[Batch1_RS_log.csv](https://github.com/RangeShifter/RangeshiftR-tutorials/files/8312537/Batch1_RS_log.csv)
[Sim1_Landscape.txt](https://github.com/RangeShifter/RangeshiftR-tutorials/files/8312545/Sim1_Landscape.txt)


Lastly, we would like to simulation 20 chromosomes with 1 locus per chromosome (NLoci = 20). Is the ArchFile properly formatted for that? 

Thanks for all of your help!

Celine

**Reply by annekathleenmalchow** [reply]
Dear Celine,
your files look ok to me and you are right: since you want to model neutral genetics there are no trait lines needed.
I have used your genetics module and architecture file with an example simulation on my computer and it works normally. 
Therefore I am a little clueless. Do you work on a Windows machine? Maybe you could make sure that your frogarch.txt is encoded in UTF-8? (We have had encoding-related problems before with file reading..) Maybe try adding a newline at the end?
Anne

**Reply by JetteReeg** [reply]
Dear Celine, I also tested your genetics module and architecture file on Windows and it worked fine. Which R version are you using? 
Jette

**Reply by celinetree** [reply]
I am using R version 4.1.2 on Mac. 

I checked the encoding and the file is indeed in UTF-8. I also tried adding a new line as Anne suggested with no luck. I am going to try and run the script on a different computer to see if that changes anything.

**Comment by celinetree** [comment]
Update: I've successfully ran the script with no issues using the ArchFile on a windows machine and produced genetic output. I attempted to update R, re-install RangeShiftR, and restart R on my mac but I still get the error above. Perhaps something with Mac is interfering with reading the archfile?

**Comment by annekathleenmalchow** [comment, marked as answer]
Hi Celine, thanks for making this effort. I am really not sure what could cause this problem. Quite a while ago we had a somewhat similar problem reading a landscape file on a Mac. It seemed to be related to the fact that the file was created on a Windows machine and with file editor used on the Mac. A workaround was to create the file in R. It might be a little far fetched, but maybe it's something you could try, as well:
```
fileConn<-file("arch.txt")
writeLines(c(paste("NChromosomes", 20),
		     paste("NLoci", do.call(paste,as.list(rep(1,20))))), fileConn)
close(fileConn)
```
I that doesn't help, we will try to re-create the error on a Mac on our side, but finding and fixing the error might take a while.
Anne

**Reply by celinetree** [reply]
Hi Anne,

Unfortunately this did not work for me either. I also tried making the text file via command line on vim with no luck.

Celine

**Reply by JetteReeg** [reply]
Hi Celine, thanks for letting us know. We were also able to recreate the error in a Mac. We will look into that problem and let you know when we fixed it. However, due to sickness and staff shortage this might take 2 weeks. We are sorry for the inconvenience and hope to fix the problem soon. 
Jette

**Reply by annekathleenmalchow** [reply]
Hi Celine,
sorry for the long wait. We have commited some changes to the RangeShiftR repository that should hopefully solve your problem.
If you would like to try it out, please re-install RangeShiftR from Github.
Anne

**Reply by celinetree** [reply]
Hello Anne,

Thank you! I finally got some time to try the code out and unfortunately I get the same error after re-installing RangeShiftR. Please let me know if I can provide any more information. 

Celine

**Reply by damariszurell** [reply]
Hi Celine,

I also tested in on my Mac and it works if the architecture file does not have an extra line after the entries. You can open the file in your RStudio editor and simply delete the empty line, then it should be fine.

**Reply by celinetree** [reply]
This worked! Thank you!
